File: algorithms/implementation/halloween-sale/halloween-sale-python/tests/test_howManyGames.py
from howManyGames import howManyGames

class TesteHowManyGames:
    
    def testeCaso1(self):
        
        p = 20 
        d = 3
        m = 6
        s = 80
        
        esperado = 6
        
        resultado = howManyGames(p, d, m, s)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        p = 20 
        d = 3
        m = 6
        s = 85
        
        esperado = 7
        
        resultado = howManyGames(p, d, m, s)
        
        assert resultado == esperado
        
    def testeCaso3(self):
        
        p = 16
        d = 2
        m = 1
        s = 9981
        
        esperado = 9917
        
        resultado = howManyGames(p, d, m, s)
        
        assert resultado == esperado
        
    def testeCaso4(self):
        
        p = 100
        d = 1
        m = 1
        s = 99
        
        esperado = 0
        
        resultado = howManyGames(p, d, m, s)
        
        assert resultado == esperado

File: algorithms/implementation/halloween-sale/halloween-sale-javascript/js/test_howManyGames.js
import {howManyGames} from '../js/howManyGames.js';
import assert from 'assert';

describe('howManyGames', function() {

    it(' o cliente pode comprar com a quantia disponível de: 6', function() {

        let p = 20;
        let d = 3;
        let m = 6;
        let s = 80;
        
        let esperado = 6;
        
        let resultado = howManyGames(p, d, m, s)

        assert.strictEqual(resultado, esperado);
    });
});


describe('howManyGames', function() {

    it(' o cliente pode comprar com a quantia disponível de: 7', function() {

        let p = 20 ;
        let d = 3;
        let m = 6;
        let s = 85;
        
        let esperado = 7;
        
        let resultado = howManyGames(p, d, m, s)

        assert.strictEqual(resultado, esperado);
    });
});

describe('howManyGames', function() {

    it(' o cliente pode comprar com a quantia disponível de: 99177', function() {

        let p = 16;
        let d = 2;
        let m = 1;
        let s = 9981;
        
        let esperado = 9917
        
        let resultado = howManyGames(p, d, m, s)

        assert.strictEqual(resultado, esperado);
    });
});